src/tls/openssl.c

```diff
@@ -434,6 +434,13 @@ static int tls_net_read(struct flb_tls_session *session,
             ERR_error_string_n(ret, err_buf, sizeof(err_buf)-1);
             flb_error("[tls] syscall error: %s", err_buf);
 
+            /* According to the documentation these are non-recoverable
```

Comment:
Any unrecoverable tls connect errors in the handshake SSL code section? - I suppose these failed connections will never make it to the keepalive available queue, so the current code is sufficient. 
https://github.com/fluent/fluent-bit/blob/ebb6ddff82d6845782643a63f001951ba82413bc/src/tls/openssl.c#L508

Reply:
You're right, errors there are caught but even then those never make it to the keepalive idle list.